src/storage.ts

```diff
@@ -93,11 +95,29 @@ export class LlamaStorage {
     };
 
     const tx = this.db.transaction(['blob', 'metadata'], 'readwrite');
+
+    let uploadTask: Promise<UploadResult> | null = null;
+    if (firebaseStorage) {
+      // Note that as bad as this looks, security-wise, there will be server-side protection
+      // against the user modifying unintended locations.
+      const storageRef = ref(firebaseStorage, 'public/' + metadata.filename);
+      const firebaseMeta: UploadMetadata = {
+        contentType: metadata.mimeType,
+        customMetadata: metadata,
+      };
+      uploadTask = uploadBytes(storageRef, file, firebaseMeta);
+    }
+
     await Promise.allSettled([
       tx.objectStore('blob').put(file, id),
       tx.objectStore('metadata').put(record, id),
       tx.done,
     ]);
+    if (uploadTask) {
+      // TODO: we should allow this in the background, not have it block (and also handle errors,
+      // etc)
+      await uploadTask;
```

Comment:
Why not initialize `uploadTask` with a resolved promise and include it in the `allSettled` call on :111?

_Not_ doing this in the background, at least for now, conceptually is a lot simpler as it'll lead to less user surprises when there are consistency concerns. In the longer term we might need to use e.g. Background Sync to retry the upload when connectivity is restored, in case stuff gets added whilst offline.

Reply:
> Why not initialize uploadTask with a resolved promise and include it in the allSettled call on :111?

Good suggestion, done.